wgpu-hal/src/lib.rs

```diff
@@ -486,9 +486,15 @@ pub trait Surface: WasmNotSendSync {
     /// the timeout will be ignored.
     ///
     /// Returns `None` on timing out.
+    ///
+    /// # Safety
+    ///
+    /// - The fence must be the same fence passed to all [`Queue::submit`]s
+    ///   that used textures acquired from this surface.
```

Comment:
This is unclear. I'm using textures acquired from this surface all the time, with many different submissions.

Reply:
This makes more sense to me now.